### PR TITLE
Refactor LocalPRG::shift(.)

### PR DESCRIPTION
**Pull request by rffrancon**
Related to issue #23 .
Please don't merge.

### CODE REVIEW

File: src/localPRG.cpp [hunk 1]
```diff
@@ -247,54 +247,30 @@ std::vector<uint32_t> LocalPRG::build_graph(const Interval &i,
...
             short_paths.pop_front();
 
-            if (n.back()->pos.get_end() == (--(prg.nodes.end()))->second->pos.get_end()) {
-                return_paths.push_back(p);
-            } else if (n.back()->pos.get_end() == p.get_end()) {
-                for (uint32_t j = 0; j != n.back()->outNodes.size(); ++j) {
-                    if (n.back()->outNodes[j]->pos.length == 0) {
-                        short_paths.push_back(p);
-                        short_paths.back().add_end_interval(n.back()->outNodes[j]->pos);
+            if (nodes.back()->pos.get_end() == (--(prg.nodes.end()))->second->pos.get_end()) {
```
Comment (rffrancon) [inline src/localPRG.cpp]: @rmcolq What does this conditional mean please?

File: src/localPRG.cpp [hunk 2]
```diff
@@ -304,11 +280,38 @@ std::vector<Path> LocalPRG::shift(Path p) const {
...
         }
     }
 
-    return return_paths;
+    return shifted_paths;
+}
+
+std::vector<Path> LocalPRG::get_extended_paths(const Path &path) const {
+    std::deque<Path> short_paths = {path};
+    std::vector<Path> k_paths = {};
```
Comment (rffrancon) [inline src/localPRG.cpp]: @rmcolq What is `k_paths`? Kmer paths?

File: src/localPRG.cpp [hunk 3]
```diff
@@ -247,54 +247,30 @@ std::vector<uint32_t> LocalPRG::build_graph(const Interval &i,
...
-        } else if (p.get_end() != (--(prg.nodes.end()))->second->pos.get_end()) {
-            for (uint32_t i = 0; i != n.back()->outNodes.size(); ++i) {
-                //exp_num_return_seqs += 1;
-                short_paths.push_back(p);
-                short_paths.back().add_end_interval(
-                        Interval(n.back()->outNodes[i]->pos.start, n.back()->outNodes[i]->pos.start));
-            }
-        }
-    }
+    Path truncated_path = path.subpath(1, path.length() - 1);
```
Comment (rffrancon) [inline src/localPRG.cpp]: Does this line shift or truncate the path? If truncate, why?

Comment (rmcolq) [reply src/localPRG.cpp]: This line takes the truncated path from position 1 along the original path to the end. I want the paths which are shifted along one position in the graph. I have the first `k-1` bases length of this desired path(s) already, so I use them and look for the next `1` base length to add to the end of the path

File: src/localPRG.cpp (hunk 2 above)
Comment (rffrancon) [reply src/localPRG.cpp]: Thanks for the response. What is `k` please?

Comment (rmcolq) [reply src/localPRG.cpp]: `k` is an input parameter and determines the length of the (w,k)-minimizers (smallest kmer in window of w consecutive kmers). I usually use `k=15`, but for other kmer-based work values are often e.g. `21`, `31`.

Comment (rffrancon) [reply src/localPRG.cpp]: Would `min_kmer_length` be an accurate name?

Comment (rmcolq) [reply src/localPRG.cpp]: `k` is `kmer_length`. All kmers have to be exactly `k` bases long. Looking again, `k_paths` are the paths of length exactly `k` and `short_paths` are the paths of length less than `k` which I'm extending to paths of length `k`.

**Comment by iqbal-lab** [comment]
hi @rffrancon !

**Pull request closed without merging**